### PR TITLE
Add settings to hold multihop entry server

### PR DESCRIPTION
**Pull request by oskirby**
This extends the `ServerData` class to hold an exit server and optional entry server, which permits the configuration of multihop tunnelling. From QML, this can be achieved by providing optional parameters to the `VPNCurrentServer.update()` or `VPNController.changeServer()` methods. When those options are omitted, only a single hop is used.

For example: `VPNController.changeServer("aq", "McMurdo Station", "ca", "North Pole")`

While we were at it, we also added a recent connections list to the `SettingsHolder` class which contains a list of up to 5 recent connections.

### CODE REVIEW

File: src/commands/commandselect.cpp [hunk 1]
```diff
@@ -21,11 +21,12 @@ CommandSelect::~CommandSelect() { MVPN_COUNT_DTOR(CommandSelect); }
 int CommandSelect::run(QStringList& tokens) {
   Q_ASSERT(!tokens.isEmpty());
   return runCommandLineApp([&]() {
-    if (tokens.length() != 2) {
+    if (tokens.length() < 2) {
```
Comment (bakulf) [inline src/commands/commandselect.cpp:24]: < 2  || > 3 ?

File: src/controller.cpp [hunk 2]
```diff
@@ -194,13 +194,9 @@ void Controller::activateInternal() {
...
   // node as the first element, and the entry node as the final entry.
   QList<Server> serverList = {server};
-  if (FeatureMultiHop::instance()->isSupported() &&
-      settingsHolder->multihopTunnel()) {
-    ServerData data;
-    Server hop = vpn->randomHop(data);
-    while (serverList.contains(hop)) {
-      hop = vpn->randomHop(data);
-    }
+  if (FeatureMultiHop::instance()->isSupported() && vpn->multihop()) {
```
Comment (bakulf) [inline src/controller.cpp:197]: Maybe we can move the `isSupported()` check in the `multihop()` method.

File: src/models/serverdata.cpp [hunk 3]
```diff
@@ -21,56 +21,101 @@ bool ServerData::fromSettings() {
...
+                     settingsHolder->currentServerCity(),
+                     settingsHolder->entryServerCountryCode(),
+                     settingsHolder->entryServerCity());
 
-  logger.debug() << m_countryCode << m_countryName << m_cityName;
+  logger.debug() << toString();
+  return true;
+}
+
+bool ServerData::fromString(const QString& data) {
```
Comment (bakulf) [inline src/models/serverdata.cpp:38]: This is just for testing, right?

Comment (bakulf) [reply src/models/serverdata.cpp:38]: Can we expose it only `#ifdef UNIT_TEST` ?

**Review by bakulf — APPROVED** [review]

File: src/controller.cpp (hunk 2 above)
Comment (oskirby) [reply src/controller.cpp:197]: I prefer to keep the stuff in the `models` directory to be just plain data containers, and to leave these kinds of conditionals in the root objects of the project. But it's not a big deal to change it either.

Comment (oskirby) [reply src/controller.cpp]: Actually, it turns out that calling `isSupported()` from within a unit test results in a segmentation fault. And I only realized that after pushing. Oops.

File: src/models/serverdata.cpp (hunk 3 above)
Comment (oskirby) [reply src/models/serverdata.cpp:38]: I'll make a note to myself to remove this function if we don't wind up using it after merging with #1603

**Pull request merged**